src/gpgme.cc

```diff
@@ -193,15 +179,14 @@ decrypted_stream_t::decrypted_stream_t(path& filename)
 
   file = open_file(filename);
   auto enc_d = setup_cipher_buffer(file);
-  dec_d = decrypt(enc_d);
-  rewind(dec_d.get());
+  auto dec_d = decrypt(enc_d).get();
 
   if (is_encrypted(enc_d)) {
     fclose(file);
     file = nullptr;
   }
 
-  set_rdbuf(new data_streambuffer_t(*dec_d.get()));
+  set_rdbuf(new data_streambuffer_t(*dec_d));
```

Comment:
I took the liberty of simplifying this code some more, does this still look correct to to @spacefrogg?

I seemed that since previously on line 197 and 204 only `.get()` was used the return value could be stored temporarly and used to call `rewind` and then pass to the `data_streambuffer_t` ctor.

What are your thoughts on this?

Reply:
Please rename `dec_d` to `dec_data`, as it is no longer of the same type as `enc_d`.

You also dropped the call to `rewind()` which is crucial before creating a new stream buffer from it.

Reply:
Apologies for having missed the accidential removal of the `rewind` call. That's why we have PR review :)